### PR TITLE
Add a discovery extension to support multiple versions of the DeltaSpike - repairs for issue #100

### PR DESCRIPTION
**Pull request by ikysil**

### CODE REVIEW

File: travisci/deltaspike-v1.x.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+# Current strategy is, for each supported minor version of DeltaSpike, to test one early point release plus
+# the most recent point release, plus some point releases which caused problems.
+DELTASPIKE_VERSIONS=(
```
Comment (seanf) [inline travisci/deltaspike-v1.x.sh:5]: "For each supported minor version of DeltaSpike, we test the most recent point release." At least, that's what seems to appear below.

**Comment by seanf** [comment]
@ikysil +1 from me once you update that comment.

**Comment by ikysil** [comment]
> So we have 3 builds of 25-30 minutes now, instead of 24 builds of about 3 mins (which finished in about 18 minutes total in the last build of master).

We can group these differently, say 1 job per MAJOR version of Weld plus 1 job for DeltaSpike tests.
When implemented this way:
* there will be 12 jobs in total - 3 (JDK) * (3 (Weld) + 1 (DeltaSpike))
* the tests for DeltaSpike will execute concurrently with the tests for Weld (using dedicated jobs for each).

The longest job will be executed for tests with Weld 2.x as it has the largest number of versions at the moment.

> That's okay, considering it's testing more DeltaSpike versions, but it can fail fast, right?

Yes, the first failing step in the script will fail the complete job.

File: travisci/deltaspike-v1.x.sh (hunk 1 above)
Comment (ikysil) [reply travisci/deltaspike-v1.x.sh:4]: Updated in https://github.com/cdi-unit/cdi-unit/commit/15f583a64588f60c9b2c889495a58e580cf7cfee

**Pull request merged**